// angular-loading-page/__ivy_ngcc__/esm5/angular-loading-page.js
import { Injectable, Optional, Injector, Directive, Input, TemplateRef, ViewContainerRef, NgModule, Component } from '@angular/core';
import { Router, NavigationStart, NavigationEnd, NavigationCancel, NavigationError } from '@angular/router';
import { Subject } from 'rxjs';
import { finalize } from 'rxjs/operators';
import { CommonModule } from '@angular/common';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';

import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
var RouteInterceptorService = /** @class */ (function () {
    function RouteInterceptorService(router) {
        var _this = this;
        this.router = router;
        this._isPendingRouteSubject = new Subject();
        if (this.router) {
            router.events.subscribe(function (event) {
                _this.routerInterceptor(event);
            });
        }
    }
    Object.defineProperty(RouteInterceptorService.prototype, "isPendingRoute", {
        get: function () {
            return this._isPendingRouteSubject.asObservable();
        },
        enumerable: true,
        configurable: true
    });
    RouteInterceptorService.prototype.routerInterceptor = function (event) {
        if (event instanceof NavigationStart) {
            this.onRouteStart();
        }
        if (event instanceof NavigationEnd || event instanceof NavigationCancel || event instanceof NavigationError) {
            this.onRouteDone();
        }
    };
    RouteInterceptorService.prototype.onRouteStart = function () {
        this._isPendingRouteSubject.next(true);
    };
    RouteInterceptorService.prototype.onRouteDone = function () {
        this._isPendingRouteSubject.next(false);
    };
RouteInterceptorService.ɵfac = function RouteInterceptorService_Factory(t) { return new (t || RouteInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc1.Router, 8)); };
RouteInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: RouteInterceptorService, factory: function (t) { return RouteInterceptorService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RouteInterceptorService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.Router, decorators: [{
                type: Optional
            }] }]; }, null); })();
    return RouteInterceptorService;
}());
RouteInterceptorService.ctorParameters = function () { return [
    { type: Router, decorators: [{ type: Optional },] },
]; };
var HttpObservableService = /** @class */ (function () {
    function HttpObservableService() {
        this._pendingRequestsCounter = 0;
        this._isPendingRequestsSubject = new Subject();
    }
    Object.defineProperty(HttpObservableService.prototype, "isPendingRequests", {
        get: function () {
            return this._isPendingRequestsSubject.asObservable();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(HttpObservableService.prototype, "pendingRequests", {
        get: function () {
            return this._pendingRequestsCounter;
        },
        enumerable: true,
        configurable: true
    });
    HttpObservableService.prototype.onRequestStart = function () {
        this._pendingRequestsCounter++;
        if (this._pendingRequestsCounter === 1) {
            this._isPendingRequestsSubject.next(true);
        }
    };
    HttpObservableService.prototype.onRequestDone = function () {
        this._pendingRequestsCounter--;
        if (this._pendingRequestsCounter === 0) {
            this._isPendingRequestsSubject.next(false);
        }
    };
HttpObservableService.ɵfac = function HttpObservableService_Factory(t) { return new (t || HttpObservableService)(); };
HttpObservableService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpObservableService, factory: function (t) { return HttpObservableService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpObservableService, [{
        type: Injectable
    }], function () { return []; }, null); })();
    return HttpObservableService;
}());
var HttpInterceptorService = /** @class */ (function () {
    function HttpInterceptorService(injector) {
        this.injector = injector;
        this.httpObservableService = injector.get(HttpObservableService);
    }
    HttpInterceptorService.prototype.intercept = function (request, next) {
        var _this = this;
        this.httpObservableService.onRequestStart();
        return next.handle(request).pipe(finalize(function () {
            _this.httpObservableService.onRequestDone();
        }));
    };
HttpInterceptorService.ɵfac = function HttpInterceptorService_Factory(t) { return new (t || HttpInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc0.Injector)); };
HttpInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpInterceptorService, factory: function (t) { return HttpInterceptorService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpInterceptorService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc0.Injector }]; }, null); })();
    return HttpInterceptorService;
}());
HttpInterceptorService.ctorParameters = function () { return [
    { type: Injector, },
]; };
function HttpInterceptorServiceFactory(injector) {
    return new HttpInterceptorService(injector);
}
var LoadingPageDirective = /** @class */ (function () {
    function LoadingPageDirective(httpObservableService, routeInterceptorService, templateRef, viewContainer) {
        this.httpObservableService = httpObservableService;
        this.routeInterceptorService = routeInterceptorService;
        this.templateRef = templateRef;
        this.viewContainer = viewContainer;
        this.hasView = false;
    }
    LoadingPageDirective.prototype.ngOnChanges = function (changes) {
        if (changes["mkLoadingPage"].currentValue) {
            this.mkLoadingPage = changes["mkLoadingPage"].currentValue;
        }
        if (changes["mkLoadingPage"].firstChange) {
            this.init();
        }
    };
    LoadingPageDirective.prototype.init = function () {
        var _this = this;
        if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingHttp) {
            this.httpObservableService.isPendingRequests.subscribe(function (value) {
                _this.isPendingRequests = value;
                _this.isVisible();
            });
        }
        if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingRoute) {
            this.routeInterceptorService.isPendingRoute.subscribe(function (value) {
                _this.isPendingRoute = value;
                _this.isVisible();
            });
        }
    };
    LoadingPageDirective.prototype.isVisible = function () {
        if (this.hasView && !this.isPendingRequests && !this.isPendingRoute) {
            this.viewContainer.clear();
            this.hasView = false;
        }
        else if (!this.hasView && (this.isPendingRequests || this.isPendingRoute)) {
            this.viewContainer.createEmbeddedView(this.templateRef);
            this.hasView = true;
        }
    };
LoadingPageDirective.ɵfac = function LoadingPageDirective_Factory(t) { return new (t || LoadingPageDirective)(ɵngcc0.ɵɵdirectiveInject(HttpObservableService), ɵngcc0.ɵɵdirectiveInject(RouteInterceptorService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
LoadingPageDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: LoadingPageDirective, selectors: [["", "mkLoadingPage", ""]], inputs: { mkLoadingPage: "mkLoadingPage" }, features: [ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingPageDirective, [{
        type: Directive,
        args: [{
                selector: '[mkLoadingPage]'
            }]
    }], function () { return [{ type: HttpObservableService }, { type: RouteInterceptorService }, { type: ɵngcc0.TemplateRef }, { type: ɵngcc0.ViewContainerRef }]; }, { mkLoadingPage: [{
            type: Input
        }] }); })();
    return LoadingPageDirective;
}());
LoadingPageDirective.ctorParameters = function () { return [
    { type: HttpObservableService, },
    { type: RouteInterceptorService, },
    { type: TemplateRef, },
    { type: ViewContainerRef, },
]; };
LoadingPageDirective.propDecorators = {
    "mkLoadingPage": [{ type: Input },],
};
var HttpInterceptorServiceExistingProvider = {
    provide: HTTP_INTERCEPTORS,
    useExisting: HttpInterceptorService,
    multi: true
};
var ɵ0 = HttpInterceptorServiceFactory;
var LoadingPageModule = /** @class */ (function () {
    function LoadingPageModule() {
    }
LoadingPageModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LoadingPageModule });
LoadingPageModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LoadingPageModule_Factory(t) { return new (t || LoadingPageModule)(); }, providers: [
        RouteInterceptorService,
        HttpObservableService,
        HttpInterceptorServiceExistingProvider,
        {
            provide: HttpInterceptorService,
            useFactory: ɵ0,
            deps: [Injector]
        }
    ], imports: [[
            CommonModule,
            HttpClientModule
        ],
        HttpClientModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LoadingPageModule, { declarations: function () { return [LoadingPageDirective]; }, imports: function () { return [CommonModule,
        HttpClientModule]; }, exports: function () { return [HttpClientModule,
        LoadingPageDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingPageModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    HttpClientModule
                ],
                declarations: [
                    LoadingPageDirective
                ],
                exports: [
                    HttpClientModule,
                    LoadingPageDirective
                ],
                providers: [
                    RouteInterceptorService,
                    HttpObservableService,
                    HttpInterceptorServiceExistingProvider,
                    {
                        provide: HttpInterceptorService,
                        useFactory: ɵ0,
                        deps: [Injector]
                    }
                ]
            }]
    }], function () { return []; }, null); })();
    return LoadingPageModule;
}());
var ChasingDotsComponent = /** @class */ (function () {
    function ChasingDotsComponent() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
ChasingDotsComponent.ɵfac = function ChasingDotsComponent_Factory(t) { return new (t || ChasingDotsComponent)(); };
ChasingDotsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ChasingDotsComponent, selectors: [["mk-chasing-dots"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-chasing-dots"], [1, "sk-child", "sk-dot1"], [1, "sk-child", "sk-dot2"]], template: function ChasingDotsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-chasing-dots[_ngcontent-%COMP%]{margin:40px auto;width:40px;height:40px;position:relative;text-align:center;-webkit-animation:2s linear infinite sk-chasingDotsRotate;animation:2s linear infinite sk-chasingDotsRotate}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{width:60%;height:60%;display:inline-block;position:absolute;top:0;border-radius:100%;-webkit-animation:2s ease-in-out infinite sk-chasingDotsBounce;animation:2s ease-in-out infinite sk-chasingDotsBounce}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-dot2[_ngcontent-%COMP%]{top:auto;bottom:0;-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-chasingDotsRotate{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes sk-chasingDotsRotate{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes sk-chasingDotsBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-chasingDotsBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsComponent, [{
        type: Component,
        args: [{
                selector: 'mk-chasing-dots',
                template: "\n    <div class=\"sk-chasing-dots\">\n      <div class=\"sk-child sk-dot1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-dot2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-chasing-dots{margin:40px auto;width:40px;height:40px;position:relative;text-align:center;-webkit-animation:2s linear infinite sk-chasingDotsRotate;animation:2s linear infinite sk-chasingDotsRotate}.sk-chasing-dots .sk-child{width:60%;height:60%;display:inline-block;position:absolute;top:0;border-radius:100%;-webkit-animation:2s ease-in-out infinite sk-chasingDotsBounce;animation:2s ease-in-out infinite sk-chasingDotsBounce}.sk-chasing-dots .sk-dot2{top:auto;bottom:0;-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-chasingDotsRotate{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes sk-chasingDotsRotate{100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@-webkit-keyframes sk-chasingDotsBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-chasingDotsBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
    return ChasingDotsComponent;
}());
ChasingDotsComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
};
var ChasingDotsModule = /** @class */ (function () {
    function ChasingDotsModule() {
    }
ChasingDotsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChasingDotsModule });
ChasingDotsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChasingDotsModule_Factory(t) { return new (t || ChasingDotsModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChasingDotsModule, { declarations: function () { return [ChasingDotsComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ChasingDotsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    ChasingDotsComponent,
                ],
                exports: [
                    ChasingDotsComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return ChasingDotsModule;
}());
var CircleComponent = /** @class */ (function () {
    function CircleComponent() {
        this.color = '#da4733';
    }
CircleComponent.ɵfac = function CircleComponent_Factory(t) { return new (t || CircleComponent)(); };
CircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CircleComponent, selectors: [["mk-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-circle"], [1, "sk-circle1", "sk-child"], [1, "sk-circle2", "sk-child"], [1, "sk-circle3", "sk-child"], [1, "sk-circle4", "sk-child"], [1, "sk-circle5", "sk-child"], [1, "sk-circle6", "sk-child"], [1, "sk-circle7", "sk-child"], [1, "sk-circle8", "sk-child"], [1, "sk-circle9", "sk-child"], [1, "sk-circle10", "sk-child"], [1, "sk-circle11", "sk-child"], [1, "sk-circle12", "sk-child"]], template: function CircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-circle[_ngcontent-%COMP%]{margin:40px auto;width:40px;height:40px;position:relative}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{width:100%;height:100%;position:absolute;left:0;top:0}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleBounceDelay;animation:1.2s ease-in-out infinite both sk-circleBounceDelay}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleComponent, [{
        type: Component,
        args: [{
                selector: 'mk-circle',
                template: "\n    <div class=\"sk-circle\">\n      <div class=\"sk-circle1 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-child\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-circle{margin:40px auto;width:40px;height:40px;position:relative}.sk-circle .sk-child{width:100%;height:100%;position:absolute;left:0;top:0}.sk-circle .sk-child>div{display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleBounceDelay;animation:1.2s ease-in-out infinite both sk-circleBounceDelay}.sk-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-circle .sk-circle2>div{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-circle .sk-circle3>div{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-circle .sk-circle4>div{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-circle .sk-circle5>div{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-circle .sk-circle6>div{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-circle .sk-circle7>div{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-circle .sk-circle8>div{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-circle .sk-circle9>div{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-circle .sk-circle10>div{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-circle .sk-circle11>div{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-circle .sk-circle12>div{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-circleBounceDelay{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return CircleComponent;
}());
CircleComponent.propDecorators = {
    "color": [{ type: Input },],
};
var CircleModule = /** @class */ (function () {
    function CircleModule() {
    }
CircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CircleModule });
CircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CircleModule_Factory(t) { return new (t || CircleModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CircleModule, { declarations: function () { return [CircleComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CircleComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    CircleComponent,
                ],
                exports: [
                    CircleComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return CircleModule;
}());
var CubeGridComponent = /** @class */ (function () {
    function CubeGridComponent() {
        this.color = '#da4733';
    }
CubeGridComponent.ɵfac = function CubeGridComponent_Factory(t) { return new (t || CubeGridComponent)(); };
CubeGridComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CubeGridComponent, selectors: [["mk-cube-grid"]], inputs: { color: "color" }, decls: 10, vars: 18, consts: [[1, "sk-cube-grid"], [1, "sk-cube", "sk-cube1"], [1, "sk-cube", "sk-cube2"], [1, "sk-cube", "sk-cube3"], [1, "sk-cube", "sk-cube4"], [1, "sk-cube", "sk-cube5"], [1, "sk-cube", "sk-cube6"], [1, "sk-cube", "sk-cube7"], [1, "sk-cube", "sk-cube8"], [1, "sk-cube", "sk-cube9"]], template: function CubeGridComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelement(6, "div", 6);
        ɵngcc0.ɵɵelement(7, "div", 7);
        ɵngcc0.ɵɵelement(8, "div", 8);
        ɵngcc0.ɵɵelement(9, "div", 9);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-cube-grid[_ngcontent-%COMP%]{width:40px;height:40px;margin:40px auto}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{width:33.33%;height:33.33%;float:left;-webkit-animation:1.3s ease-in-out infinite sk-cubeGridScaleDelay;animation:1.3s ease-in-out infinite sk-cubeGridScaleDelay}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube1[_ngcontent-%COMP%]{-webkit-animation-delay:.2s;animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{-webkit-animation-delay:.4s;animation-delay:.4s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{-webkit-animation-delay:.1s;animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube5[_ngcontent-%COMP%]{-webkit-animation-delay:.2s;animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube6[_ngcontent-%COMP%]{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube7[_ngcontent-%COMP%]{-webkit-animation-delay:0ms;animation-delay:0ms}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube8[_ngcontent-%COMP%]{-webkit-animation-delay:.1s;animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube9[_ngcontent-%COMP%]{-webkit-animation-delay:.2s;animation-delay:.2s}@-webkit-keyframes sk-cubeGridScaleDelay{0%,100%,70%{-webkit-transform:scale3D(1,1,1);transform:scale3D(1,1,1)}35%{-webkit-transform:scale3D(0,0,1);transform:scale3D(0,0,1)}}@keyframes sk-cubeGridScaleDelay{0%,100%,70%{-webkit-transform:scale3D(1,1,1);transform:scale3D(1,1,1)}35%{-webkit-transform:scale3D(0,0,1);transform:scale3D(0,0,1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridComponent, [{
        type: Component,
        args: [{
                selector: 'mk-cube-grid',
                template: "\n    <div class=\"sk-cube-grid\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube5\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube6\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube7\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube8\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube9\" [style.background-color]=\"color\"></div>\n    </div>",
                styles: [".sk-cube-grid{width:40px;height:40px;margin:40px auto}.sk-cube-grid .sk-cube{width:33.33%;height:33.33%;float:left;-webkit-animation:1.3s ease-in-out infinite sk-cubeGridScaleDelay;animation:1.3s ease-in-out infinite sk-cubeGridScaleDelay}.sk-cube-grid .sk-cube1{-webkit-animation-delay:.2s;animation-delay:.2s}.sk-cube-grid .sk-cube2{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-cube-grid .sk-cube3{-webkit-animation-delay:.4s;animation-delay:.4s}.sk-cube-grid .sk-cube4{-webkit-animation-delay:.1s;animation-delay:.1s}.sk-cube-grid .sk-cube5{-webkit-animation-delay:.2s;animation-delay:.2s}.sk-cube-grid .sk-cube6{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-cube-grid .sk-cube7{-webkit-animation-delay:0ms;animation-delay:0ms}.sk-cube-grid .sk-cube8{-webkit-animation-delay:.1s;animation-delay:.1s}.sk-cube-grid .sk-cube9{-webkit-animation-delay:.2s;animation-delay:.2s}@-webkit-keyframes sk-cubeGridScaleDelay{0%,100%,70%{-webkit-transform:scale3D(1,1,1);transform:scale3D(1,1,1)}35%{-webkit-transform:scale3D(0,0,1);transform:scale3D(0,0,1)}}@keyframes sk-cubeGridScaleDelay{0%,100%,70%{-webkit-transform:scale3D(1,1,1);transform:scale3D(1,1,1)}35%{-webkit-transform:scale3D(0,0,1);transform:scale3D(0,0,1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return CubeGridComponent;
}());
CubeGridComponent.propDecorators = {
    "color": [{ type: Input },],
};
var CubeGridModule = /** @class */ (function () {
    function CubeGridModule() {
    }
CubeGridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CubeGridModule });
CubeGridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CubeGridModule_Factory(t) { return new (t || CubeGridModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CubeGridModule, { declarations: function () { return [CubeGridComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CubeGridComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    CubeGridComponent,
                ],
                exports: [
                    CubeGridComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return CubeGridModule;
}());
var DoubleBounceComponent = /** @class */ (function () {
    function DoubleBounceComponent() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
DoubleBounceComponent.ɵfac = function DoubleBounceComponent_Factory(t) { return new (t || DoubleBounceComponent)(); };
DoubleBounceComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DoubleBounceComponent, selectors: [["mk-double-bounce"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-double-bounce"], [1, "sk-child", "sk-double-bounce1"], [1, "sk-child", "sk-double-bounce2"]], template: function DoubleBounceComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-double-bounce[_ngcontent-%COMP%]{width:40px;height:40px;position:relative;margin:40px auto}.sk-double-bounce[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{width:100%;height:100%;border-radius:50%;opacity:.6;position:absolute;top:0;left:0;-webkit-animation:2s ease-in-out infinite sk-doubleBounce;animation:2s ease-in-out infinite sk-doubleBounce}.sk-double-bounce[_ngcontent-%COMP%]   .sk-double-bounce2[_ngcontent-%COMP%]{-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-doubleBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-doubleBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceComponent, [{
        type: Component,
        args: [{
                selector: 'mk-double-bounce',
                template: "\n    <div class=\"sk-double-bounce\">\n      <div class=\"sk-child sk-double-bounce1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-double-bounce2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-double-bounce{width:40px;height:40px;position:relative;margin:40px auto}.sk-double-bounce .sk-child{width:100%;height:100%;border-radius:50%;opacity:.6;position:absolute;top:0;left:0;-webkit-animation:2s ease-in-out infinite sk-doubleBounce;animation:2s ease-in-out infinite sk-doubleBounce}.sk-double-bounce .sk-double-bounce2{-webkit-animation-delay:-1s;animation-delay:-1s}@-webkit-keyframes sk-doubleBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-doubleBounce{0%,100%{-webkit-transform:scale(0);transform:scale(0)}50%{-webkit-transform:scale(1);transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
    return DoubleBounceComponent;
}());
DoubleBounceComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
};
var DoubleBounceModule = /** @class */ (function () {
    function DoubleBounceModule() {
    }
DoubleBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DoubleBounceModule });
DoubleBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DoubleBounceModule_Factory(t) { return new (t || DoubleBounceModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DoubleBounceModule, { declarations: function () { return [DoubleBounceComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DoubleBounceComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    DoubleBounceComponent,
                ],
                exports: [
                    DoubleBounceComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return DoubleBounceModule;
}());
var FadingCircleComponent = /** @class */ (function () {
    function FadingCircleComponent() {
        this.color = '#da4733';
    }
FadingCircleComponent.ɵfac = function FadingCircleComponent_Factory(t) { return new (t || FadingCircleComponent)(); };
FadingCircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FadingCircleComponent, selectors: [["mk-fading-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-fading-circle"], [1, "sk-circle1", "sk-circle"], [1, "sk-circle2", "sk-circle"], [1, "sk-circle3", "sk-circle"], [1, "sk-circle4", "sk-circle"], [1, "sk-circle5", "sk-circle"], [1, "sk-circle6", "sk-circle"], [1, "sk-circle7", "sk-circle"], [1, "sk-circle8", "sk-circle"], [1, "sk-circle9", "sk-circle"], [1, "sk-circle10", "sk-circle"], [1, "sk-circle11", "sk-circle"], [1, "sk-circle12", "sk-circle"]], template: function FadingCircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-fading-circle[_ngcontent-%COMP%]{margin:40px auto;width:40px;height:40px;position:relative}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%]{width:100%;height:100%;position:absolute;left:0;top:0}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleFadeDelay;animation:1.2s ease-in-out infinite both sk-circleFadeDelay}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}@keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleComponent, [{
        type: Component,
        args: [{
                selector: 'mk-fading-circle',
                template: "\n    <div class=\"sk-fading-circle\">\n      <div class=\"sk-circle1 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-fading-circle{margin:40px auto;width:40px;height:40px;position:relative}.sk-fading-circle .sk-circle{width:100%;height:100%;position:absolute;left:0;top:0}.sk-fading-circle .sk-circle>div{content:'';display:block;margin:0 auto;width:15%;height:15%;border-radius:100%;-webkit-animation:1.2s ease-in-out infinite both sk-circleFadeDelay;animation:1.2s ease-in-out infinite both sk-circleFadeDelay}.sk-fading-circle .sk-circle2{-webkit-transform:rotate(30deg);transform:rotate(30deg)}.sk-fading-circle .sk-circle3{-webkit-transform:rotate(60deg);transform:rotate(60deg)}.sk-fading-circle .sk-circle4{-webkit-transform:rotate(90deg);transform:rotate(90deg)}.sk-fading-circle .sk-circle5{-webkit-transform:rotate(120deg);transform:rotate(120deg)}.sk-fading-circle .sk-circle6{-webkit-transform:rotate(150deg);transform:rotate(150deg)}.sk-fading-circle .sk-circle7{-webkit-transform:rotate(180deg);transform:rotate(180deg)}.sk-fading-circle .sk-circle8{-webkit-transform:rotate(210deg);transform:rotate(210deg)}.sk-fading-circle .sk-circle9{-webkit-transform:rotate(240deg);transform:rotate(240deg)}.sk-fading-circle .sk-circle10{-webkit-transform:rotate(270deg);transform:rotate(270deg)}.sk-fading-circle .sk-circle11{-webkit-transform:rotate(300deg);transform:rotate(300deg)}.sk-fading-circle .sk-circle12{-webkit-transform:rotate(330deg);transform:rotate(330deg)}.sk-fading-circle .sk-circle2>div{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-fading-circle .sk-circle3>div{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-fading-circle .sk-circle4>div{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-fading-circle .sk-circle5>div{-webkit-animation-delay:-.8s;animation-delay:-.8s}.sk-fading-circle .sk-circle6>div{-webkit-animation-delay:-.7s;animation-delay:-.7s}.sk-fading-circle .sk-circle7>div{-webkit-animation-delay:-.6s;animation-delay:-.6s}.sk-fading-circle .sk-circle8>div{-webkit-animation-delay:-.5s;animation-delay:-.5s}.sk-fading-circle .sk-circle9>div{-webkit-animation-delay:-.4s;animation-delay:-.4s}.sk-fading-circle .sk-circle10>div{-webkit-animation-delay:-.3s;animation-delay:-.3s}.sk-fading-circle .sk-circle11>div{-webkit-animation-delay:-.2s;animation-delay:-.2s}.sk-fading-circle .sk-circle12>div{-webkit-animation-delay:-.1s;animation-delay:-.1s}@-webkit-keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}@keyframes sk-circleFadeDelay{0%,100%,39%{opacity:0}40%{opacity:1}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return FadingCircleComponent;
}());
FadingCircleComponent.propDecorators = {
    "color": [{ type: Input },],
};
var FadingCircleModule = /** @class */ (function () {
    function FadingCircleModule() {
    }
FadingCircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FadingCircleModule });
FadingCircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FadingCircleModule_Factory(t) { return new (t || FadingCircleModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FadingCircleModule, { declarations: function () { return [FadingCircleComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FadingCircleComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    FadingCircleComponent,
                ],
                exports: [
                    FadingCircleComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return FadingCircleModule;
}());
var FoldingCubeComponent = /** @class */ (function () {
    function FoldingCubeComponent() {
        this.color = '#da4733';
    }
FoldingCubeComponent.ɵfac = function FoldingCubeComponent_Factory(t) { return new (t || FoldingCubeComponent)(); };
FoldingCubeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FoldingCubeComponent, selectors: [["mk-folding-cube"]], inputs: { color: "color" }, decls: 9, vars: 8, consts: [[1, "sk-folding-cube"], [1, "sk-cube1", "sk-cube"], [1, "sk-cube2", "sk-cube"], [1, "sk-cube4", "sk-cube"], [1, "sk-cube3", "sk-cube"]], template: function FoldingCubeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-folding-cube[_ngcontent-%COMP%]{margin:40px auto;width:40px;height:40px;position:relative;-webkit-transform:rotateZ(45deg);transform:rotateZ(45deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{float:left;width:50%;height:50%;position:relative;-webkit-transform:scale(1.1);transform:scale(1.1)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{content:'';position:absolute;top:0;left:0;width:100%;height:100%;-webkit-animation:2.4s linear infinite both sk-foldCubeAngle;animation:2.4s linear infinite both sk-foldCubeAngle;-webkit-transform-origin:100% 100%;transform-origin:100% 100%}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{-webkit-transform:scale(1.1) rotateZ(90deg);transform:scale(1.1) rotateZ(90deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{-webkit-transform:scale(1.1) rotateZ(180deg);transform:scale(1.1) rotateZ(180deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{-webkit-transform:scale(1.1) rotateZ(270deg);transform:scale(1.1) rotateZ(270deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:.6s;animation-delay:.6s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{-webkit-animation-delay:.9s;animation-delay:.9s}@-webkit-keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeComponent, [{
        type: Component,
        args: [{
                selector: 'mk-folding-cube',
                template: "\n    <div class=\"sk-folding-cube\">\n      <div class=\"sk-cube1 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube2 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube4 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube3 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-folding-cube{margin:40px auto;width:40px;height:40px;position:relative;-webkit-transform:rotateZ(45deg);transform:rotateZ(45deg)}.sk-folding-cube .sk-cube{float:left;width:50%;height:50%;position:relative;-webkit-transform:scale(1.1);transform:scale(1.1)}.sk-folding-cube .sk-cube>div{content:'';position:absolute;top:0;left:0;width:100%;height:100%;-webkit-animation:2.4s linear infinite both sk-foldCubeAngle;animation:2.4s linear infinite both sk-foldCubeAngle;-webkit-transform-origin:100% 100%;transform-origin:100% 100%}.sk-folding-cube .sk-cube2{-webkit-transform:scale(1.1) rotateZ(90deg);transform:scale(1.1) rotateZ(90deg)}.sk-folding-cube .sk-cube3{-webkit-transform:scale(1.1) rotateZ(180deg);transform:scale(1.1) rotateZ(180deg)}.sk-folding-cube .sk-cube4{-webkit-transform:scale(1.1) rotateZ(270deg);transform:scale(1.1) rotateZ(270deg)}.sk-folding-cube .sk-cube2>div{-webkit-animation-delay:.3s;animation-delay:.3s}.sk-folding-cube .sk-cube3>div{-webkit-animation-delay:.6s;animation-delay:.6s}.sk-folding-cube .sk-cube4>div{-webkit-animation-delay:.9s;animation-delay:.9s}@-webkit-keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}@keyframes sk-foldCubeAngle{0%,10%{-webkit-transform:perspective(140px) rotateX(-180deg);transform:perspective(140px) rotateX(-180deg);opacity:0}25%,75%{-webkit-transform:perspective(140px) rotateX(0);transform:perspective(140px) rotateX(0);opacity:1}100%,90%{-webkit-transform:perspective(140px) rotateY(180deg);transform:perspective(140px) rotateY(180deg);opacity:0}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return FoldingCubeComponent;
}());
FoldingCubeComponent.propDecorators = {
    "color": [{ type: Input },],
};
var FoldingCubeModule = /** @class */ (function () {
    function FoldingCubeModule() {
    }
FoldingCubeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FoldingCubeModule });
FoldingCubeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FoldingCubeModule_Factory(t) { return new (t || FoldingCubeModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FoldingCubeModule, { declarations: function () { return [FoldingCubeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FoldingCubeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    FoldingCubeComponent,
                ],
                exports: [
                    FoldingCubeComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return FoldingCubeModule;
}());
var MaterialbarComponent = /** @class */ (function () {
    function MaterialbarComponent() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
        this.colorThree = '#fdba2c';
    }
MaterialbarComponent.ɵfac = function MaterialbarComponent_Factory(t) { return new (t || MaterialbarComponent)(); };
MaterialbarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MaterialbarComponent, selectors: [["mk-material-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo", colorThree: "colorThree" }, decls: 4, vars: 6, consts: [[1, "material-bar"], [1, "bar"]], template: function MaterialbarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 1);
        ɵngcc0.ɵɵelement(3, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorThree);
    } }, styles: [".material-bar[_ngcontent-%COMP%]{position:relative;width:100%;height:4px;background-color:#fdba2c}.bar[_ngcontent-%COMP%]{content:\"\";display:inline;position:absolute;width:0;height:100%;left:50%;text-align:center}.bar[_ngcontent-%COMP%]:nth-child(1){-webkit-animation:3s linear infinite loading;animation:3s linear infinite loading}.bar[_ngcontent-%COMP%]:nth-child(2){-webkit-animation:3s linear 1s infinite loading;animation:3s linear 1s infinite loading}.bar[_ngcontent-%COMP%]:nth-child(3){-webkit-animation:3s linear 2s infinite loading;animation:3s linear 2s infinite loading}@-webkit-keyframes loading{from{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}@keyframes loading{from{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialbarComponent, [{
        type: Component,
        args: [{
                selector: 'mk-material-bar',
                template: "\n  <div class=\"material-bar\">\n    <div class=\"bar\" [style.background-color]=\"colorOne\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorTwo\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorThree\"></div>\n  </div>",
                styles: [".material-bar{position:relative;width:100%;height:4px;background-color:#fdba2c}.bar{content:\"\";display:inline;position:absolute;width:0;height:100%;left:50%;text-align:center}.bar:nth-child(1){-webkit-animation:3s linear infinite loading;animation:3s linear infinite loading}.bar:nth-child(2){-webkit-animation:3s linear 1s infinite loading;animation:3s linear 1s infinite loading}.bar:nth-child(3){-webkit-animation:3s linear 2s infinite loading;animation:3s linear 2s infinite loading}@-webkit-keyframes loading{from{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}@keyframes loading{from{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }], colorThree: [{
            type: Input
        }] }); })();
    return MaterialbarComponent;
}());
MaterialbarComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
    "colorThree": [{ type: Input },],
};
var MaterialBarModule = /** @class */ (function () {
    function MaterialBarModule() {
    }
MaterialBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MaterialBarModule });
MaterialBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MaterialBarModule_Factory(t) { return new (t || MaterialBarModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MaterialBarModule, { declarations: function () { return [MaterialbarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MaterialbarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialBarModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    MaterialbarComponent,
                ],
                exports: [
                    MaterialbarComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return MaterialBarModule;
}());
var PulseComponent = /** @class */ (function () {
    function PulseComponent() {
        this.color = '#da4733';
    }
PulseComponent.ɵfac = function PulseComponent_Factory(t) { return new (t || PulseComponent)(); };
PulseComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PulseComponent, selectors: [["mk-pulse"]], inputs: { color: "color" }, decls: 1, vars: 2, consts: [[1, "sk-spinner", "sk-spinner-pulse"]], template: function PulseComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-spinner-pulse[_ngcontent-%COMP%]{width:40px;height:40px;margin:40px auto;background-color:#333;border-radius:100%;-webkit-animation:1s ease-in-out infinite sk-pulseScaleOut;animation:1s ease-in-out infinite sk-pulseScaleOut}@-webkit-keyframes sk-pulseScaleOut{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}@keyframes sk-pulseScaleOut{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseComponent, [{
        type: Component,
        args: [{
                selector: 'mk-pulse',
                template: "<div class=\"sk-spinner sk-spinner-pulse\" [style.background-color]=\"color\"></div>",
                styles: [".sk-spinner-pulse{width:40px;height:40px;margin:40px auto;background-color:#333;border-radius:100%;-webkit-animation:1s ease-in-out infinite sk-pulseScaleOut;animation:1s ease-in-out infinite sk-pulseScaleOut}@-webkit-keyframes sk-pulseScaleOut{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}@keyframes sk-pulseScaleOut{0%{-webkit-transform:scale(0);transform:scale(0)}100%{-webkit-transform:scale(1);transform:scale(1);opacity:0}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return PulseComponent;
}());
PulseComponent.propDecorators = {
    "color": [{ type: Input },],
};
var PulseModule = /** @class */ (function () {
    function PulseModule() {
    }
PulseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PulseModule });
PulseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PulseModule_Factory(t) { return new (t || PulseModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PulseModule, { declarations: function () { return [PulseComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [PulseComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    PulseComponent,
                ],
                exports: [
                    PulseComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return PulseModule;
}());
var SlidingBarComponent = /** @class */ (function () {
    function SlidingBarComponent() {
        this.colorOne = '#ddd';
        this.colorTwo = '#3c8dbc';
    }
SlidingBarComponent.ɵfac = function SlidingBarComponent_Factory(t) { return new (t || SlidingBarComponent)(); };
SlidingBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SlidingBarComponent, selectors: [["mk-sliding-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 2, vars: 4, consts: [[1, "sliding-bar"]], template: function SlidingBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sliding-bar[_ngcontent-%COMP%]{height:4px;width:100%;position:relative;overflow:hidden}.sliding-bar[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{display:block;position:absolute;content:\"\";left:-200px;width:200px;height:4px;background-color:#2980b9;-webkit-animation:2s linear infinite loading;animation:2s linear infinite loading}@-webkit-keyframes loading{from{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}@keyframes loading{from{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarComponent, [{
        type: Component,
        args: [{
                selector: 'mk-sliding-bar',
                template: "<div class=\"sliding-bar\" [style.background-color]=\"colorOne\"><div [style.background-color]=\"colorTwo\"></div></div>",
                styles: [".sliding-bar{height:4px;width:100%;position:relative;overflow:hidden}.sliding-bar>div{display:block;position:absolute;content:\"\";left:-200px;width:200px;height:4px;background-color:#2980b9;-webkit-animation:2s linear infinite loading;animation:2s linear infinite loading}@-webkit-keyframes loading{from{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}@keyframes loading{from{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
    return SlidingBarComponent;
}());
SlidingBarComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
};
var SlidingBarModule = /** @class */ (function () {
    function SlidingBarModule() {
    }
SlidingBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SlidingBarModule });
SlidingBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SlidingBarModule_Factory(t) { return new (t || SlidingBarModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SlidingBarModule, { declarations: function () { return [SlidingBarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SlidingBarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    SlidingBarComponent,
                ],
                exports: [
                    SlidingBarComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return SlidingBarModule;
}());
var ThreeBounceComponent = /** @class */ (function () {
    function ThreeBounceComponent() {
        this.colorOne = '#00a65a';
        this.colorTwo = '#fdba2c';
        this.colorThree = '#da4733';
    }
ThreeBounceComponent.ɵfac = function ThreeBounceComponent_Factory(t) { return new (t || ThreeBounceComponent)(); };
ThreeBounceComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ThreeBounceComponent, selectors: [["mk-three-bounce"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo", colorThree: "colorThree" }, decls: 4, vars: 6, consts: [[1, "sk-three-bounce"], [1, "sk-child", "sk-bounce1"], [1, "sk-child", "sk-bounce2"], [1, "sk-child", "sk-bounce3"]], template: function ThreeBounceComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorThree);
    } }, styles: [".sk-three-bounce[_ngcontent-%COMP%]{margin:40px auto;width:80px;text-align:center}.sk-three-bounce[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{width:20px;height:20px;background-color:#333;border-radius:100%;display:inline-block;-webkit-animation:1.4s ease-in-out infinite both sk-three-bounce;animation:1.4s ease-in-out infinite both sk-three-bounce}.sk-three-bounce[_ngcontent-%COMP%]   .sk-bounce1[_ngcontent-%COMP%]{-webkit-animation-delay:-.32s;animation-delay:-.32s}.sk-three-bounce[_ngcontent-%COMP%]   .sk-bounce2[_ngcontent-%COMP%]{-webkit-animation-delay:-.16s;animation-delay:-.16s}@-webkit-keyframes sk-three-bounce{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-three-bounce{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ThreeBounceComponent, [{
        type: Component,
        args: [{
                selector: 'mk-three-bounce',
                template: "\n    <div class=\"sk-three-bounce\">\n       <div class=\"sk-child sk-bounce1\" [style.background-color]=\"colorOne\"></div>\n       <div class=\"sk-child sk-bounce2\" [style.background-color]=\"colorTwo\"></div>\n       <div class=\"sk-child sk-bounce3\" [style.background-color]=\"colorThree\"></div>\n     </div>",
                styles: [".sk-three-bounce{margin:40px auto;width:80px;text-align:center}.sk-three-bounce .sk-child{width:20px;height:20px;background-color:#333;border-radius:100%;display:inline-block;-webkit-animation:1.4s ease-in-out infinite both sk-three-bounce;animation:1.4s ease-in-out infinite both sk-three-bounce}.sk-three-bounce .sk-bounce1{-webkit-animation-delay:-.32s;animation-delay:-.32s}.sk-three-bounce .sk-bounce2{-webkit-animation-delay:-.16s;animation-delay:-.16s}@-webkit-keyframes sk-three-bounce{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}@keyframes sk-three-bounce{0%,100%,80%{-webkit-transform:scale(0);transform:scale(0)}40%{-webkit-transform:scale(1);transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }], colorThree: [{
            type: Input
        }] }); })();
    return ThreeBounceComponent;
}());
ThreeBounceComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
    "colorThree": [{ type: Input },],
};
var ThreeBounceModule = /** @class */ (function () {
    function ThreeBounceModule() {
    }
ThreeBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ThreeBounceModule });
ThreeBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ThreeBounceModule_Factory(t) { return new (t || ThreeBounceModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ThreeBounceModule, { declarations: function () { return [ThreeBounceComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ThreeBounceComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ThreeBounceModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    ThreeBounceComponent,
                ],
                exports: [
                    ThreeBounceComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return ThreeBounceModule;
}());
var WanderingCubesComponent = /** @class */ (function () {
    function WanderingCubesComponent() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
WanderingCubesComponent.ɵfac = function WanderingCubesComponent_Factory(t) { return new (t || WanderingCubesComponent)(); };
WanderingCubesComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WanderingCubesComponent, selectors: [["mk-wandering-cubes"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-wandering-cubes"], [1, "sk-cube", "sk-cube1"], [1, "sk-cube", "sk-cube2"]], template: function WanderingCubesComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-wandering-cubes[_ngcontent-%COMP%]{margin:40px auto;width:40px;height:40px;position:relative}.sk-wandering-cubes[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{width:10px;height:10px;position:absolute;top:0;left:0;-webkit-animation:1.8s ease-in-out -1.8s infinite both sk-wanderingCube;animation:1.8s ease-in-out -1.8s infinite both sk-wanderingCube}.sk-wandering-cubes[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{-webkit-animation-delay:-.9s;animation-delay:-.9s}@-webkit-keyframes sk-wanderingCube{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%{-webkit-transform:translateX(30px) rotate(-90deg) scale(.5);transform:translateX(30px) rotate(-90deg) scale(.5)}50%{-webkit-transform:translateX(30px) translateY(30px) rotate(-179deg);transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{-webkit-transform:translateX(30px) translateY(30px) rotate(-180deg);transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{-webkit-transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5);transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}100%{-webkit-transform:rotate(-360deg);transform:rotate(-360deg)}}@keyframes sk-wanderingCube{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%{-webkit-transform:translateX(30px) rotate(-90deg) scale(.5);transform:translateX(30px) rotate(-90deg) scale(.5)}50%{-webkit-transform:translateX(30px) translateY(30px) rotate(-179deg);transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{-webkit-transform:translateX(30px) translateY(30px) rotate(-180deg);transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{-webkit-transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5);transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}100%{-webkit-transform:rotate(-360deg);transform:rotate(-360deg)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WanderingCubesComponent, [{
        type: Component,
        args: [{
                selector: 'mk-wandering-cubes',
                template: "\n    <div class=\"sk-wandering-cubes\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-wandering-cubes{margin:40px auto;width:40px;height:40px;position:relative}.sk-wandering-cubes .sk-cube{width:10px;height:10px;position:absolute;top:0;left:0;-webkit-animation:1.8s ease-in-out -1.8s infinite both sk-wanderingCube;animation:1.8s ease-in-out -1.8s infinite both sk-wanderingCube}.sk-wandering-cubes .sk-cube2{-webkit-animation-delay:-.9s;animation-delay:-.9s}@-webkit-keyframes sk-wanderingCube{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%{-webkit-transform:translateX(30px) rotate(-90deg) scale(.5);transform:translateX(30px) rotate(-90deg) scale(.5)}50%{-webkit-transform:translateX(30px) translateY(30px) rotate(-179deg);transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{-webkit-transform:translateX(30px) translateY(30px) rotate(-180deg);transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{-webkit-transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5);transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}100%{-webkit-transform:rotate(-360deg);transform:rotate(-360deg)}}@keyframes sk-wanderingCube{0%{-webkit-transform:rotate(0);transform:rotate(0)}25%{-webkit-transform:translateX(30px) rotate(-90deg) scale(.5);transform:translateX(30px) rotate(-90deg) scale(.5)}50%{-webkit-transform:translateX(30px) translateY(30px) rotate(-179deg);transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{-webkit-transform:translateX(30px) translateY(30px) rotate(-180deg);transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{-webkit-transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5);transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}100%{-webkit-transform:rotate(-360deg);transform:rotate(-360deg)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
    return WanderingCubesComponent;
}());
WanderingCubesComponent.propDecorators = {
    "colorOne": [{ type: Input },],
    "colorTwo": [{ type: Input },],
};
var WanderingCubesModule = /** @class */ (function () {
    function WanderingCubesModule() {
    }
WanderingCubesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WanderingCubesModule });
WanderingCubesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WanderingCubesModule_Factory(t) { return new (t || WanderingCubesModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WanderingCubesModule, { declarations: function () { return [WanderingCubesComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WanderingCubesComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WanderingCubesModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    WanderingCubesComponent,
                ],
                exports: [
                    WanderingCubesComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return WanderingCubesModule;
}());
var WaveComponent = /** @class */ (function () {
    function WaveComponent() {
        this.color = '#da4733';
    }
WaveComponent.ɵfac = function WaveComponent_Factory(t) { return new (t || WaveComponent)(); };
WaveComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WaveComponent, selectors: [["mk-wave"]], inputs: { color: "color" }, decls: 6, vars: 10, consts: [[1, "sk-wave"], [1, "sk-rect", "sk-rect1"], [1, "sk-rect", "sk-rect2"], [1, "sk-rect", "sk-rect3"], [1, "sk-rect", "sk-rect4"], [1, "sk-rect", "sk-rect5"]], template: function WaveComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-wave[_ngcontent-%COMP%]{margin:40px auto;width:50px;height:40px;text-align:center;font-size:10px}.sk-wave[_ngcontent-%COMP%]   .sk-rect[_ngcontent-%COMP%]{height:100%;width:6px;display:inline-block;-webkit-animation:1.2s ease-in-out infinite sk-waveStretchDelay;animation:1.2s ease-in-out infinite sk-waveStretchDelay}.sk-wave[_ngcontent-%COMP%]   .sk-rect1[_ngcontent-%COMP%]{-webkit-animation-delay:-1.2s;animation-delay:-1.2s}.sk-wave[_ngcontent-%COMP%]   .sk-rect2[_ngcontent-%COMP%]{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect3[_ngcontent-%COMP%]{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect4[_ngcontent-%COMP%]{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-wave[_ngcontent-%COMP%]   .sk-rect5[_ngcontent-%COMP%]{-webkit-animation-delay:-.8s;animation-delay:-.8s}@-webkit-keyframes sk-waveStretchDelay{0%,100%,40%{-webkit-transform:scaleY(.4);transform:scaleY(.4)}20%{-webkit-transform:scaleY(1);transform:scaleY(1)}}@keyframes sk-waveStretchDelay{0%,100%,40%{-webkit-transform:scaleY(.4);transform:scaleY(.4)}20%{-webkit-transform:scaleY(1);transform:scaleY(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveComponent, [{
        type: Component,
        args: [{
                selector: 'mk-wave',
                template: "\n    <div class=\"sk-wave\">\n      <div class=\"sk-rect sk-rect1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect5\" [style.background-color]=\"color\"></div>\n    </div>",
                styles: [".sk-wave{margin:40px auto;width:50px;height:40px;text-align:center;font-size:10px}.sk-wave .sk-rect{height:100%;width:6px;display:inline-block;-webkit-animation:1.2s ease-in-out infinite sk-waveStretchDelay;animation:1.2s ease-in-out infinite sk-waveStretchDelay}.sk-wave .sk-rect1{-webkit-animation-delay:-1.2s;animation-delay:-1.2s}.sk-wave .sk-rect2{-webkit-animation-delay:-1.1s;animation-delay:-1.1s}.sk-wave .sk-rect3{-webkit-animation-delay:-1s;animation-delay:-1s}.sk-wave .sk-rect4{-webkit-animation-delay:-.9s;animation-delay:-.9s}.sk-wave .sk-rect5{-webkit-animation-delay:-.8s;animation-delay:-.8s}@-webkit-keyframes sk-waveStretchDelay{0%,100%,40%{-webkit-transform:scaleY(.4);transform:scaleY(.4)}20%{-webkit-transform:scaleY(1);transform:scaleY(1)}}@keyframes sk-waveStretchDelay{0%,100%,40%{-webkit-transform:scaleY(.4);transform:scaleY(.4)}20%{-webkit-transform:scaleY(1);transform:scaleY(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
    return WaveComponent;
}());
WaveComponent.propDecorators = {
    "color": [{ type: Input },],
};
var WaveModule = /** @class */ (function () {
    function WaveModule() {
    }
WaveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WaveModule });
WaveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WaveModule_Factory(t) { return new (t || WaveModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WaveModule, { declarations: function () { return [WaveComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WaveComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    WaveComponent,
                ],
                exports: [
                    WaveComponent
                ]
            }]
    }], function () { return []; }, null); })();
    return WaveModule;
}());

export { LoadingPageModule, LoadingPageDirective, HttpInterceptorService, HttpInterceptorServiceFactory, HttpObservableService, RouteInterceptorService, ChasingDotsModule, ChasingDotsComponent, CircleModule, CircleComponent, CubeGridModule, CubeGridComponent, DoubleBounceModule, DoubleBounceComponent, FadingCircleModule, FadingCircleComponent, FoldingCubeModule, FoldingCubeComponent, MaterialBarModule, MaterialbarComponent, PulseModule, PulseComponent, SlidingBarModule, SlidingBarComponent, ThreeBounceModule, ThreeBounceComponent, WanderingCubesModule, WanderingCubesComponent, WaveModule, WaveComponent };
