// angular-admin-lte/esm2015/lib/layout/header/header.component.js
import { __decorate, __metadata } from "tslib";
import { AfterViewInit, Component, ContentChild, ElementRef, Input, NgZone, OnDestroy, Renderer2, TemplateRef, ViewChild } from '@angular/core';
import { LayoutStore } from '../layout.store';
import { HeaderService } from './header.service';
import { removeListeners, removeSubscriptions } from '../../helpers';
/**
 * Header Logo
 */
let HeaderLogoComponent = class HeaderLogoComponent {
};
__decorate([
    ViewChild('templateRef', { static: true }),
    __metadata("design:type", TemplateRef)
], HeaderLogoComponent.prototype, "templateRef", void 0);
HeaderLogoComponent = __decorate([
    Component({
        selector: 'mk-layout-header-logo',
        template: '<ng-template #templateRef><ng-content></ng-content></ng-template>'
    })
], HeaderLogoComponent);
export { HeaderLogoComponent };
/**
 * Header Logo Mini
 */
let HeaderLogoMiniComponent = class HeaderLogoMiniComponent {
};
__decorate([
    ViewChild('templateRef', { static: true }),
    __metadata("design:type", TemplateRef)
], HeaderLogoMiniComponent.prototype, "templateRef", void 0);
HeaderLogoMiniComponent = __decorate([
    Component({
        selector: 'mk-layout-header-logo-mini',
        template: '<ng-template #templateRef><ng-content></ng-content></ng-template>'
    })
], HeaderLogoMiniComponent);
export { HeaderLogoMiniComponent };
/**
 * Header
 */
let HeaderComponent = class HeaderComponent {
    /**
     * @method constructor
     * @param layoutStore [description]
     * @param ngZone      [description]
     * @param renderer2   [description]
     * @param elementRef   [description]
     * @param headerService   [description]
     */
    constructor(layoutStore, ngZone, renderer2, elementRef, headerService) {
        this.layoutStore = layoutStore;
        this.ngZone = ngZone;
        this.renderer2 = renderer2;
        this.elementRef = elementRef;
        this.headerService = headerService;
        this.listeners = [];
        this.subscriptions = [];
        this.isSidebarLeftToggle = true;
        this.isSidebarRightToggle = true;
        this.logoLink = '/';
    }
    /**
     * @method ngAfterViewInit
     */
    ngAfterViewInit() {
        this.headerService.elementRef = this.headerElement;
        if (this.sidebarLeftToggleElement) {
            this.subscriptions.push(this.layoutStore.isSidebarLeftCollapsed.subscribe((value) => {
                this.isSidebarLeftCollapsed = value;
            }));
            this.ngZone.runOutsideAngular(() => {
                this.listeners.push(this.renderer2.listen(this.sidebarLeftToggleElement.nativeElement, 'click', (event) => {
                    event.preventDefault();
                    this.layoutStore.sidebarLeftCollapsed(!this.isSidebarLeftCollapsed);
                }));
            });
        }
        if (this.sidebarRightToggleElement) {
            this.subscriptions.push(this.layoutStore.isSidebarRightCollapsed.subscribe((value) => {
                this.isSidebarRightCollapsed = value;
            }));
            this.ngZone.runOutsideAngular(() => {
                this.listeners.push(this.renderer2.listen(this.sidebarRightToggleElement.nativeElement, 'click', (event) => {
                    event.preventDefault();
                    this.layoutStore.sidebarRightCollapsed(!this.isSidebarRightCollapsed);
                }));
            });
        }
    }
    /**
     * @method ngOnDestroy
     */
    ngOnDestroy() {
        this.listeners = removeListeners(this.listeners);
        this.subscriptions = removeSubscriptions(this.subscriptions);
    }
};
HeaderComponent.ctorParameters = () => [
    { type: LayoutStore },
    { type: NgZone },
    { type: Renderer2 },
    { type: ElementRef },
    { type: HeaderService }
];
__decorate([
    Input(),
    __metadata("design:type", Object)
], HeaderComponent.prototype, "isSidebarLeftToggle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], HeaderComponent.prototype, "isSidebarRightToggle", void 0);
__decorate([
    Input(),
    __metadata("design:type", Object)
], HeaderComponent.prototype, "logoLink", void 0);
__decorate([
    ContentChild(HeaderLogoComponent, /* TODO: add static flag */ {}),
    __metadata("design:type", HeaderLogoComponent)
], HeaderComponent.prototype, "headerLogoComponent", void 0);
__decorate([
    ContentChild(HeaderLogoMiniComponent, /* TODO: add static flag */ {}),
    __metadata("design:type", HeaderLogoMiniComponent)
], HeaderComponent.prototype, "headerLogoMiniComponent", void 0);
__decorate([
    ViewChild('headerElement', { static: true }),
    __metadata("design:type", ElementRef)
], HeaderComponent.prototype, "headerElement", void 0);
__decorate([
    ViewChild('sidebarLeftToggleElement'),
    __metadata("design:type", ElementRef)
], HeaderComponent.prototype, "sidebarLeftToggleElement", void 0);
__decorate([
    ViewChild('sidebarRightToggleElement'),
    __metadata("design:type", ElementRef)
], HeaderComponent.prototype, "sidebarRightToggleElement", void 0);
HeaderComponent = __decorate([
    Component({
        selector: 'mk-layout-header',
        template: "<header #headerElement class=\"main-header\">\n  <a [routerLink]=\"logoLink\" class=\"logo\">\n    <span class=\"logo-mini\"><ng-template [ngTemplateOutlet]=\"headerLogoMiniComponent?.templateRef\"></ng-template></span>\n    <span class=\"logo-lg\"><ng-template [ngTemplateOutlet]=\"headerLogoComponent?.templateRef\"></ng-template></span>\n  </a>\n  <nav class=\"navbar navbar-static-top\">\n    <a *ngIf=\"isSidebarLeftToggle\" #sidebarLeftToggleElement href=\"#\" class=\"sidebar-toggle\">\n      <span class=\"sr-only\">Toggle navigation</span>\n      <span class=\"icon-bar\"></span>\n      <span class=\"icon-bar\"></span>\n      <span class=\"icon-bar\"></span>\n    </a>\n    <div *ngIf=\"isSidebarRightToggle\" class=\"sidebar-right-toggle\">\n      <a #sidebarRightToggleElement href=\"#\"><i class=\"fa fa-gears\"></i></a>\n    </div>\n    <ng-content></ng-content>\n  </nav>\n</header>\n",
        styles: [":host{display:block}.sidebar-right-toggle{float:right}.sidebar-right-toggle a{padding:15px;display:block;line-height:20px;cursor:pointer}"]
    }),
    __metadata("design:paramtypes", [LayoutStore,
        NgZone,
        Renderer2,
        ElementRef,
        HeaderService])
], HeaderComponent);
export { HeaderComponent };
